paasta_tools/kubernetes_tools.py

```diff
@@ -2568,3 +2596,33 @@ def to_node_label(label: str) -> str:
     }:
         return f"yelp.com/{label}"
     return label
+
+
+def get_all_service_accounts(
+    kube_client: KubeClient, namespace: str,
+) -> Sequence[V1ServiceAccount]:
+    return kube_client.core.list_namespaced_service_account(namespace=namespace).items
+
+
+_RE_NORMALIZE_IAM_ROLE = re.compile(r"[^0-9a-zA-Z]+")
+
+
+def create_or_find_service_account_name(
+    iam_role: str, namespace: str = "paasta"
+) -> str:
+    kube_client = KubeClient()
+    sa_name = "paasta--" + _RE_NORMALIZE_IAM_ROLE.sub("-", iam_role)
```

Comment:
if serviceaccount is a namespaced object, do we need `paasta--` prefix?

Reply:
you're right, let's remove it.

Reply:
Actually
```
NAME                                                     SECRETS   AGE
cassandra-operator                                       1         406d
default                                                  1         407d
flink-operator                                           1         407d
kafka-operator                                           1         302d
nrtsearch-operator                                       1         93d
paasta--arn-aws-iam-528741615426-role-flink-kubernetes   1         5h32m
zzzzzzz                                                  1         3d4h
```
so, even it's namespaced, there could be other service accounts in the `paasta` namespace not related to paasta tools, so we probably better separate them somehow?

Reply:
ok, going to merge this PR, we can change it in a next PR.